src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java

```diff
@@ -138,10 +140,22 @@ private LoadFlowResult runAc(Network network, LoadFlowParameters parameters, Ope
                                                                           parameters.isDistributedSlack() && (parameters.getBalanceType() == LoadFlowParameters.BalanceType.PROPORTIONAL_TO_LOAD || parameters.getBalanceType() == LoadFlowParameters.BalanceType.PROPORTIONAL_TO_CONFORM_LOAD) && parametersExt.isLoadPowerFactorConstant(),
                                                                           parameters.isDc(),
                                                                           acParameters.getNetworkParameters().isBreakers());
-                result.getNetwork().updateState(updateParameters);
 
-                // zero or low impedance branch flows computation
-                computeZeroImpedanceFlows(result.getNetwork(), LoadFlowModel.AC);
+                if (parametersExt.isNetworkCacheEnabled()) {
+                    // to avoid in case of LF with transformer voltage control, that cache is invalidated because of
+                    // tap position update
+                    NetworkCache.INSTANCE.pauseListener(network, () -> {
+                        result.getNetwork().updateState(updateParameters);
+
+                        // zero or low impedance branch flows computation
```

Comment:
Why the code line 
```` computeZeroImpedanceFlows(result.getNetwork(), LoadFlowModel.AC); ````
is in the if and in the else, as it is not concerned by the listener pause?

Reply:
It sets iidm switch p q so it is concerned

Reply:
Yes you right, it is in the runnable, sorry !